java/common/common-task/src/main/java/sleeper/task/common/RunCompactionTasks.java

```diff
@@ -165,7 +166,7 @@ private static void launchTasks(
         String fargateVersion = instanceProperties.get(FARGATE_VERSION);
         String launchType = instanceProperties.get(COMPACTION_ECS_LAUNCHTYPE);
         TaskOverride override = createOverride(List.of(instanceProperties.get(CONFIG_BUCKET)), COMPACTION_CONTAINER_NAME);
-        NetworkConfiguration networkConfiguration = networkConfig(instanceProperties.getList(SUBNETS));
+        NetworkConfiguration networkConfiguration = networkConfig(instanceProperties);
         String defUsed = (launchType.equalsIgnoreCase("FARGATE")) ? fargateTaskDefinition : ec2TaskDefinition;
```

Comment:
Can we move all this stuff into createRunTaskRequest and just pass in the instance properties? That way we'll only need to check the launch type once, and only create the Fargate-specific parts for the Fargate launch type. As it is, it's pretty difficult to tell that the network configuration is just for Fargate.